datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -945,7 +323,6 @@ impl PartialEq<dyn Any> for InListExpr {
                 self.expr.eq(&x.expr)
                     && expr_list_eq_any_order(&self.list, &x.list)
                     && self.negated == x.negated
```

Comment:
```suggestion
                    && self.negated == x.negated
                    // since self.static_filter is derived from self.list, not necessary to check here
```